coral-component-colorpicker/src/scripts/ColorUtil.js

```diff
@@ -0,0 +1,81 @@
+/**
+ * Copyright 2021 Adobe. All rights reserved.
+ * This file is licensed to you under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License. You may obtain a copy
+ * of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under
+ * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+ * OF ANY KIND, either express or implied. See the License for the specific language
+ * governing permissions and limitations under the License.
+ */
+import { TinyColor } from '@ctrl/tinycolor';
+
+class ColorUtil {
+
+  constructor() {}
+  /**
+   Utility function to convert a color to Hsl string.
+
+   @param {String} hue
+   The hue value  which is preserved in color conversion.
+   @param {String} colorString
+   The color string
+   */
+  toHslString(hue, colorString) {
+    const color =  new TinyColor(colorString);
+    const hueExp = /(^hs[v|l]a?\()\d{1,3}/; 
+    return color.toHslString().replace(hueExp, `$1${hue}`);
+  }
+
+  /**
+   Utility function to convert a color to Hsv string.
+
+   @param {String} hue
+   The hue value  which is preserved in color conversion.
+   @param {String} colorString
+   The color string
+   */
+  toHsvString(hue, colorString) {
+    const color =  new TinyColor(colorString);
+    const hueExp = /(^hs[v|l]a?\()\d{1,3}/; 
+    return color.toHsvString().replace(hueExp, `$1${hue}`);
+  }
+
+  /**
+   Utility function to extract  h,s and v from a color string.
+   @param {String} colorString
+   The color string
+   */
+  extractHsv(colorString) {
+    const exp = /^hsva?\((\d{1,3}),\s*(\d{1,3}%),\s*(\d{1,3}%)/;
```

Comment:
Please add a single line comment explaining the regex. would be useful for future.